### DISCUSSION TITLE
Software licence Management

### DISCUSSION THREAD

**Discussion by robbagithub** [netbox-community/netbox#9504] [Ideas]
Hi, 
Controversial as Netbox doesn't support the concept of software licensing. However I'm wondering if another model is able to be added to support the recording of software licensing details? 

Only reason I ask is that there are many different products out there that claim to manage Software licences however they are over engineered and there seems to be no simple solution for this issue.

Other than a spreadsheet does anyone have suggestions of the best way forward as I can only assume that netbox would not have SAM in it's list of targets!

Regards,
Rob.

**Comment by mtinberg** [comment]
> Controversial as Netbox doesn't support the concept of software licensing

Netbox is licensed, under the terms of the Apache License v2 2004
https://github.com/netbox-community/netbox/blob/develop/LICENSE.txt
or maybe I misunderstood the statement, without a license you wouldn't have
permission to the software

> Other than a spreadsheet does anyone have suggestions of the best way forward as I can only assume that netbox would not have SAM in it's list of targets!

Netbox doesn't have anything specific for software license management as part of its data model, but it does have Comments on most data types as well as Custom Fields, or even Journals, which can be used to track the data you would put in a free-form spreadsheet, directly in Netbox.  Any data view can be exported as CSV and these fields can be read/write through the REST API or used with custom code in reports/scripts.

You should be able to build a process for documenting this kind of data using the features that Netbox does have, is there anything specific you had in mind?

**Reply by robbagithub** [reply]
Thanks for responding @mtinberg. The problem is that we manage 150 ish line items of specific software that we run on items across our infrastructure such as Red Hat, IBM AIX, Oracle DB, Microsoft server products, ESXi etc...

I've tried extending the custom fields of platforms to cope which is the closest object to that requirement, however the custom fields aren't searchable so it makes that difficult.  I've looked at many different products which are available and either all of hosted on line which we would prefer not to have or are not extendable in the way that Netbox is.  I think we'll be back to confluence/sharepoint/excel to model it :)

**Reply by julianze** [reply]
You can use inventory items for Software license, not perfect, but better than custom fields or comments

**Reply by robbagithub** [reply]
Hi @julianze 

We've implemented via the Platform object. The reason we stayed away from Inventory is that it **has** to be associated with an object. So with the example of virtual OS they can't be pinned down to a fixed individual hardware device as the VMs can move around the cluster.   I've attached the list of custom fields that we added. When viewed in platform in pages they present well. Not perfect at all, but better than a spreadsheet.

![cf](https://user-images.githubusercontent.com/22147754/173734247-c6166824-1778-4acd-9a0e-6d447f360de6.png)

**Comment by mtinberg** [comment]
Netbox custom fields are searchable, or at least you can search for Device or other records that have a particular custom field with a particular value

eg. /dcim/devices/?cf_software_ms=somevalue

would search all the devices for a custom field called "software_ms" that had the value of "somevalue", and these can be combined, eg. /dcim/devices/?cf_software_ms=foo&cf_software_ms=bar.  I'm not sure if there are more search options than exact match like __gt, __lt, __isw, __ic, etc.

I guess how you use these fields depends on what you are expecting to get out of it, are you using it for config automation, using the field to drop a license file or run an enabling command as part of provisioning, just as a way to collect and track the data manually, or as a datastore for some other front-end management tool you write for your site.  Netbox is going to be Device-centric though, if you just want a list of all the different software license relationships you have and aren't starting from how those tie to individual Devices, then maybe an external spreadsheet (or a Netbox plugin with its own data model and tables) would make the most sense.




—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________

**Comment by aalmarshad** [comment]
I would love to see this feature

**Comment by AndMrzv** [comment]
There is a software lifecycle plugin https://github.com/ICTU/netbox_slm which supports keeping records of installations, maybe it will fit your needs.
